xla/hlo/tools/convert.cc

```diff
@@ -0,0 +1,104 @@
+/* Copyright 2025 The OpenXLA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <stdio.h>
+
+#include <string>
+#include <vector>
+
+#include "xla/service/hlo_proto_util.h"
+#include "xla/tools/multihost_hlo_runner/functional_hlo_runner.h"
+#include "xla/tsl/util/command_line_flags.h"
+#include "tsl/platform/init_main.h"
+#include "tsl/platform/logging.h"
+#include "tsl/platform/protobuf.h"
+
+const std::string kUsage = R"(
+Reads an HLO module and outputs it in the requested format.
+)";
+
+namespace xla {
+namespace tools {
+
+absl::Status RealMain(absl::string_view input_file,
+                      absl::string_view output_file,
+                      absl::string_view input_format_str,
+                      absl::string_view output_format_str) {
+  InputFormat input_format;
+  OutputFormat output_format;
+  if (!input_format_str.empty()) {
+    if (std::string error;
+        !AbslParseFlag(input_format_str, &input_format, &error)) {
+      return absl::InternalError("Failed parsing input format: " + error);
+    }
+  }
+  if (!output_format_str.empty()) {
+    if (std::string error;
+        !AbslParseFlag(output_format_str, &output_format, &error)) {
+      return absl::InternalError("Failed parsing output format: " + error);
+    }
+  }
+  TF_ASSIGN_OR_RETURN(
+      const auto module_and_arguments,
+      FunctionalHloRunner::LoadHloModuleAndArguments(input_file, input_format));
```

Comment:
Can you maybe extract this functionality into a separate target and let FunctionalHloRunner and this tool depend on it? It seems a bit hacky to depend on FunctionalHloRunner if in fact we don't want to run anything.

Reply:
Agree. I wanted to move HLO-related functions from the runner in a later PR but will do it now.